### PR TITLE
[CI] Add type checks

### PR DESCRIPTION
**Pull request by odesenfans**
Added a new step to check types using mypy. Fixed a few typing
issues, mostly related to protobufs:
* Fixed typing issues in daemon.py
* Recompiled a type hint file for the crypto protobuf spec
* Removed some useless type ignores
* Added new type ignores where needed (some mismatches between
  classes and their protobuf representation).

### CODE REVIEW

**Comment by odesenfans** [comment]
Depends on #51.

File: setup.py [hunk 1]
```diff
@@ -11,13 +11,17 @@
...
+        "pytest-cov>=2.7.1,<3.0.0",
+        "pytest>=4.6.3,<5.0.0",
+        "types-protobuf",
+    ],
+    "lint": [
         "black>=19.3b0",
-        "isort>=4.3.21",
         "flake8>=3.7.7,<4.0.0",
+        "isort>=4.3.21",
+        "mypy-protobuf>=1.16",
```
Comment (mhchia) [inline setup.py:24]: Should `mypy-protobuf` be in the same place as `mypy`?

**Review by mhchia — APPROVED** [review]
Looks good!

File: setup.py (hunk 1 above)
Comment (odesenfans) [reply setup.py:24]: No because it's not used during the testing phase, it's purely used to generate protobuf annotations.

**Pull request merged**